### PR TITLE
Allow host loopback

### PR DESCRIPTION
**Pull request by mrmarkuz**
Add `--network=slirp4netns:allow_host_loopback=true` to allow opening documents again.
NethServer/dev#7197

### CODE REVIEW

**Comment by stephdl** [comment]
I am testing you pull request and I check if it works correctly, thank @mrmarkuz

**Comment by stephdl** [comment]
hello

I failed to use collabora

image


Does  it work on your side ?

**Comment by mrmarkuz** [comment]
Yes, it works on my side even without valid certificate. Can you reach the collabora url from the nextcloud-app container?
```
[nextcloud3@ns8rockytest state]$ podman exec -ti nextcloud-app curl -k https://collabora.ns8rockytest.com
OK
```

**Comment by stephdl** [comment]
it seems yes

```
[nextcloud1@r3-pve state]$  podman exec -ti nextcloud-app curl -k https://collabora.rocky9-pve3.org
OK
[nextcloud1@r3-pve state]$
```

**Comment by mrmarkuz** [comment]
Is the collabora url accessible from your clients web browser? Maybe wrong DNS on client?

**Comment by stephdl** [comment]
image


I use rockylinux 9.5, dns is are workable

```
[collabora1@r3-pve state]$ cat /etc/rocky-release
Rocky Linux release 9.5 (Blue Onyx)
```

**Comment by stephdl** [comment]
ok fun, it does not work in my lab, but also with rockylinux 9.4

```
[root@R1-pve ~]# cat /etc/rocky-release
Rocky Linux release 9.4 (Blue Onyx)
```

**Comment by stephdl** [comment]
thank for you patience, chrome  Version 131.0.6778.85 (Build officiel) (64 bits) was the guilty guy, I get on rockylinux 9.4 collabora workable

**Comment by mrmarkuz** [comment]
The warning is OK, you may add hosts/networks to allow in advanced settings for more security but it's not mandatory.
grafik

**Comment by stephdl** [comment]
image
no luck :p

**Comment by mrmarkuz** [comment]
This is the exactly the error you get with podman 5 without the fix.
Did you try to restart the collabora service?

**Comment by stephdl** [comment]
hold on @mrmarkuz, I tested again with real certificate on a digital ocean and it works, also only with `--network=slirp4netns`

no need to allow the localhost

**Comment by stephdl** [comment]
When to Use Which Option
Use --network=slirp4netns if you want the container to be isolated and avoid accidental access to host resources.
Use --network=slirp4netns:allow_host_loopback=true if the container needs to interact with host-based services during development or testing.

[complete output of chatgpt](https://chatgpt.com/share/6751cbc9-23a4-8011-9489-c60fc0ee70f9)

**Comment by stephdl** [comment]
I would propose we go first to --network=slirp4netns

File: imageroot/systemd/user/collabora.service [hunk 1]
```diff
@@ -11,6 +11,7 @@ ExecStartPre=/bin/rm -f %t/collabora.pid %t/collabora.ctr-id
 ExecStart=/usr/bin/podman run --conmon-pidfile %t/collabora.pid \
     --cidfile %t/collabora.ctr-id --cgroups=no-conmon \
     --replace -d --name  collabora --cap-add MKNOD \
+    --network=slirp4netns:allow_host_loopback=true \
```
Comment (stephdl) [inline imageroot/systemd/user/collabora.service:14]: --network=slirp4netns \

**Review by stephdl — CHANGES_REQUESTED** [review]

File: imageroot/systemd/user/collabora.service (hunk 1 above)
Comment (mrmarkuz) [reply imageroot/systemd/user/collabora.service]: You're right. I removed it.

**Review by mrmarkuz — COMMENTED** [review]
I removed it.

**Pull request merged**